R/plotSeries.R

```diff
@@ -0,0 +1,147 @@
+#' object.
+#'
+#' @param abund_values
+#' A single character value for selecting the
+#'   \code{\link[SummarizedExperiment:SummarizedExperiment-class]{assay}}
+#'   to be plotted.
+#'
+#' @param x_axis
+#'  A single character value for selecting the column from 
+#'  \code{\link[SummarizedExperiment:SummarizedExperiment-class]{ColData}} 
+#'  that will specify values of x-axis. 
+#'  
+#' @param x_axis
+#'  A single character value for selecting the taxa from 
+#'  \code{\link[SummarizedExperiment:SummarizedExperiment-class]{rownames}}. 
+#'  This parameter specifies taxa whose abundances will be plotted.
+#'  
+#'  @param rank a single character defining a taxonomic rank, that is used to
+#'  agglomerate the data. Must be a value of \code{taxonomicRanks()} function.
+#'
+#' @details
+#' Creates a plot where core taxa is presented against time. ###############################
+#'
+#'
+#' @references
+#' Add here.#####################################
+#'
+#' @return
+#' Returns plot###############################
+#'
+#' @name plotSeries
+#' @export
+#'
+#' @author Leo Lahti and Tuomas Borman. Contact: \url{microbiome.github.io}
+#'
+#' @examples
+#' data(esophagus)
+#' x <- esophagus
+#'
+NULL
+
+#' @rdname plotSeries
+#' @export
+setGeneric("plotSeries", signature = c("x"),
+           function(x,
+                    abund_values = NULL,
+                    x_axis = NULL,
+                    y_axis = NULL,
+                    rank = NULL)
+               standardGeneric("plotSeries"))
+
+
+#' @rdname plotSeries
+#' @export
+setMethod("plotSeries", signature = c(x = "SummarizedExperiment"),
+          function(x,
+                   abund_values = NULL,
+                   x_axis = NULL,
+                   y_axis = NULL,
+                   rank = NULL){
+              
+              ###################### Input check #######################
+              # Check abund_values
+              .check_abund_values(abund_values, x)
+              
+              # Check x_axis
+              if( !(x_axis %in% names(colData(x))) ){
+                  stop("'x_axis' must be a name of column of colData(x)", call. = FALSE)
+              }
+              
+              # Check y_axis
+              # If y_axis is not null, user has specified it
+              if (!is.null(y_axis)){
+                  if(!(y_axis %in% rownames(x)) ){
+                      stop("'y_axis' must be in rownames(x).", call. = FALSE)
+                  }
+              }
+              
+              
+              # Check rank
+              # IF rank is not null, it has been specified by user
+              if( !is.null(rank) ){
+                  if(!.is_non_empty_string(rank)){
+                      stop("'rank' must be an non empty single character value,
+                           and it must be one of taxonomyRanks(x).",
+                           call. = FALSE)
+                  }
+                  .check_taxonomic_rank(rank, x)
+              }
```

Comment:
Here two if clause follow each other with the same check. Merge them and move the above the check for `y`. See comment above